### PR TITLE
xrootd: add rmdir request

### PR DESCRIPTION
**Pull request by EgorMatirov**
Updates go-hep/hep#170.
Fixes go-hep/hep#237.

### CODE REVIEW

File: xrootd/client/file_test.go [hunk 1]
```diff
@@ -50,7 +52,14 @@ func testFile_CloseVerify(t *testing.T, addr string) {
 
 	fs := client.FS()
 
-	file, err := fs.Open(context.Background(), "/tmp/test.txt", xrdfs.OpenModeOwnerWrite, xrdfs.OpenOptionsOpenUpdate|xrdfs.OpenOptionsOpenAppend)
+	fileParent, err := tempdir(client, "/tmp/", "xrd-test-close-verify")
```
Comment (sbinet) [inline xrootd/client/file_test.go]: what about `s/fileParent/dir/` ?

File: xrootd/client/filesystem_test.go [hunk 2]
```diff
@@ -6,12 +6,28 @@ package client
...
 	"testing"
 
+	"github.com/pkg/errors"
 	"go-hep.org/x/hep/xrootd/xrdfs"
 )
 
+func tempdir(client *Client, dir, prefix string) (name string, err error) {
+	fs := client.FS()
+	for i := 0; i < 10; i++ {
+		name = path.Join(dir, prefix+"-"+strconv.Itoa(rand.Int()))
```
Comment (sbinet) [inline xrootd/client/filesystem_test.go]: I think I was really hinting at directly using `ioutil.TempDir` :)
ie:
```go
func tempdir(client *Client, dir, prefix string) (string, error) {
	name, err := ioutil.TempDir(dir, prefix)
	if err != nil {
		return "", err
	}
	os.RemoveAll(name)

	err = fs.MkdirAll(ctx, name, ...)
	if err != nil {
		return "", errors.Errorf(...)
	}
	return name, nil
}
```

File: xrootd/client/filesystem.go [hunk 3]
```diff
@@ -91,6 +92,13 @@ func (fs *fileSystem) MkdirAll(ctx context.Context, path string, perm xrdfs.Open
 	return err
 }
 
+// RemoveFile removes a directory.
```
Comment (sbinet) [inline xrootd/client/filesystem.go]: `s/RemoveFile/RemoveDir/`

**Comment by EgorMatirov** [comment]
I was waiting for https://github.com/go-hep/hep/pull/241 to rebase and don't bother with failing CI.

Thanks!

**Pull request merged**